packages/explorer/src/app/interact/Form.tsx

```diff
@@ -0,0 +1,103 @@
+"use client";
+
+import { Coins, Eye, Send } from "lucide-react";
+import { Abi, AbiFunction } from "viem";
+import { useDeferredValue, useState } from "react";
+import { Input } from "../../components/ui/Input";
+import { useHashState } from "../../hooks/useHash";
+import { cn } from "../../lib/utils";
+import { FunctionField } from "./FunctionField";
+
+type Props = {
+  data: {
+    abi: Abi;
+  };
+};
+
+export function Form({ data }: Props) {
+  const [hash] = useHashState();
+  const [filterValue, setFilterValue] = useState("");
+  const deferredFilterValue = useDeferredValue(filterValue);
+  const abiFunctions = data.abi.filter(
+    (abi) => (abi as AbiFunction).type === "function",
+  ) as AbiFunction[];
```

Comment:
rather than casting, you could also do narrowing with

```ts
const abiFunctions = data.abi.filter((item): item is AbiFunction => item.type === "function");
```

Reply:
oh this is much better with type predicate